File: app/src/main/java/net/opendasharchive/openarchive/projects/AddProjectActivity.java
package net.opendasharchive.openarchive.projects;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.MenuItem;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import net.opendasharchive.openarchive.R;
import net.opendasharchive.openarchive.db.Space;
import net.opendasharchive.openarchive.onboarding.FirstStartActivity;

public class AddProjectActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_project);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle("");
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);


    }

    public void onNewProjectClicked (View view) {

        Space space = Space.getCurrentSpace();
        if (space != null) {
            Intent intent = new Intent(this, CreateNewProjectActivity.class);
            startActivityForResult(intent, 1000);
        }
        else
        {
            finish();
            startActivity(new Intent(this, FirstStartActivity.class));
        }
    }

    public void onBrowseProjects (View view) {

        Space space = Space.getCurrentSpace();
        if (space != null) {
            Intent intent = new Intent(this, BrowseProjectsActivity.class);
            startActivityForResult(intent, 1001);
        }
        else
        {
            finish();
            startActivity(new Intent(this, FirstStartActivity.class));
        }

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {

            // Respond to the action bar's Up/Home button
            case android.R.id.home:
                //NavUtils.navigateUpFromSameTask(this);
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (resultCode == RESULT_OK) {
            setResult(RESULT_OK);
            finish();
        }
    }
}

File: app/src/main/java/net/opendasharchive/openarchive/features/projects/AddProjectActivity.kt
package net.opendasharchive.openarchive.features.projects

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.view.WindowManager
import androidx.activity.result.contract.ActivityResultContracts
import net.opendasharchive.openarchive.databinding.ActivityAddProjectBinding
import net.opendasharchive.openarchive.db.Space
import net.opendasharchive.openarchive.features.core.BaseActivity
import net.opendasharchive.openarchive.features.media.browse.BrowseProjectsActivity
import net.opendasharchive.openarchive.features.onboarding.SpaceSetupActivity

class AddProjectActivity : BaseActivity() {

    private lateinit var mBinding: ActivityAddProjectBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        window.setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE)

        mBinding = ActivityAddProjectBinding.inflate(layoutInflater)
        setContentView(mBinding.root)

        mBinding.newProject.setOnClickListener {
            setProject(false)
        }

        mBinding.browseProjects.setOnClickListener {
            setProject(true)
        }

        setSupportActionBar(mBinding.toolbar)
        supportActionBar?.title = ""
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        // We cannot browse the Internet Archive. Directly forward to creating a project,
        // as it doesn't make sense to show a one-option menu.
        if (Space.getCurrent()?.tType == Space.Type.INTERNET_ARCHIVE) {
            mBinding.browseProjects.visibility = View.GONE

            finish()
            setProject(false)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    private val mResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
        if (it.resultCode == RESULT_OK) {
            setResult(RESULT_OK)
            finish()
        }
    }

    private fun setProject(browse: Boolean) {
        if (Space.getCurrent() == null) {
            finish()
            startActivity(Intent(this, SpaceSetupActivity::class.java))

            return
        }

        mResultLauncher.launch(Intent(this,
            if (browse) BrowseProjectsActivity::class.java else CreateNewProjectActivity::class.java))
    }
}